### PR TITLE
restores add_gas(), assert_gas(), and thermal_energy() as wrapper procs

### PR DESCRIPTION
**Pull request by duncathan**
these are for use *outside* of the atmos module, for the convenience of non-atmos coders

@vuonojenmustaturska lmk if you have any concerns with how i did it

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
scrubbers, filters and miners are probably performance-sensitive to some degree. Or would be if we used miners. I don't really see a need to roll back the changes on existing code, just let the procs be an option to coders.

**Comment by Spaceinaba** [comment]
GAS GAS GAS

**Comment by optimumtact** [comment]
I disagree, either make macros or keep this unrolled.

Convenience is not an option when touching atmos code, you must be reasonably code fluent.

**Comment by duncathan** [comment]
I have been working for a long time to make atmos code more accessible and while I understand your point I really do not agree. Adding a gas to a mixture is extremely basic stuff - there is no reason we should enforce arcane-looking macros when a wrapper proc like this is perfectly fine outside of the module.

**Comment by Cyberboss** [comment]
I agree with citrus, we already have similar defines to these elsewhere in the codebase, I think we can keep it as is

**Comment by duncathan** [comment]
Defines like this are very different from things like the math defines. These macros are unintuitive in usage, left outside of the gas mixture file, and worst of all are not even implemented consistently. Internally, where the performance benefits outweigh these issues, the macros are justified. But the performance enhancements are negligible outside of gas mixtures and environmental atmos. Citing the status quo of atmos being unapproachable is a meaningless argument when its maintainer is working towards changing that.

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -37,14 +37,24 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	reaction_results = new
 
 //listmos procs
+//use the macros in performance intensive areas. for their definitions, refer to code/__DEFINES/atmospherics.dm
 
-// The following procs used to live here: thermal_energy(), assert_gas() and add_gas(). They have been moved into defines in code/__DEFINES/atmospherics.dm
+	//assert_gas(gas_id) - used to guarantee that the gas list for this id exists in gas_mixture.gases.
+	//Must be used before adding to a gas. May be used before reading from a gas.
+/datum/gas_mixture/proc/assert_gas(gas_id)
+	ASSERT_GAS(gas_id, src)
```
[the remark this answers is not in the document]
Comment (duncathan) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm]: I mean I could copy and paste the definition but I don't know if that's better

**Pull request merged**